src/did_resolve.rs

```diff
@@ -20,112 +24,192 @@ use crate::error::Error;
 use crate::one_or_many::OneOrMany;
 
+/// Media type for JSON.
 pub const TYPE_JSON: &str = "application/json";
+/// Media type for JSON-LD.
 pub const TYPE_LD_JSON: &str = "application/ld+json";
+/// Media type for a [DID Document in JSON
+/// representation](https://www.w3.org/TR/did-core/#application-did-json).
 pub const TYPE_DID_JSON: &str = "application/did+json";
+/// Media type for a [DID Document in JSON-LD
+/// representation](https://www.w3.org/TR/did-core/#application-did-ld-json).
 pub const TYPE_DID_LD_JSON: &str = "application/did+ld+json";
+/// Pseudo-media-type used when returning a URL from [DID URL
+/// dereferencing](https://w3c-ccg.github.io/did-resolution/#dereferencing-algorithm).
 pub const TYPE_URL: &str = "text/url";
+/// [`invalidDid`](https://www.w3.org/TR/did-spec-registries/#invaliddid) error value for DID
+/// Resolution / DID URL Dereferencing.
 pub const ERROR_INVALID_DID: &str = "invalidDid";
+/// [`invalidDidUrl`](https://www.w3.org/TR/did-spec-registries/#invaliddidurl) error value for DID URL Dereferencing.
 pub const ERROR_INVALID_DID_URL: &str = "invalidDidUrl";
+/// `unauthorized` error for DID Resolution / DID URL Dereferencing.
 pub const ERROR_UNAUTHORIZED: &str = "unauthorized";
+/// [`notFound`](https://www.w3.org/TR/did-spec-registries/#notfound) error value for DID URL Dereferencing.
 pub const ERROR_NOT_FOUND: &str = "notFound";
+/// `methodNotSupported` error value for DID Resolution / DID URL Dereferencing.
 pub const ERROR_METHOD_NOT_SUPPORTED: &str = "methodNotSupported";
+/// [`representationNotSupported`](https://www.w3.org/TR/did-spec-registries/#representationnotsupported) error value for DID URL Dereferencing.
 pub const ERROR_REPRESENTATION_NOT_SUPPORTED: &str = "representationNotSupported";
+/// Media type expected for a [DID Resolution Result][ResolutionResult].
 pub const TYPE_DID_RESOLUTION: &str =
     "application/ld+json;profile=\"https://w3id.org/did-resolution\";charset=utf-8";
 
-// Maximum number of DID controllers to resolve at once
+/// Maximum level of recursion when following DID controller links.
 pub const MAX_CONTROLLERS: usize = 100;
 
+/// [Metadata structure](https://www.w3.org/TR/did-core/#metadata-structure) "for DID resolution,
+/// DID URL dereferencing, and other DID-related processes"
 #[derive(Debug, Serialize, Deserialize, Clone)]
 #[serde(untagged)]
 pub enum Metadata {
+    /// [String](https://infra.spec.whatwg.org/#string)
     String(String),
+    /// A [map](https://infra.spec.whatwg.org/#maps) of properties for a metadata structure.
     Map(HashMap<String, Metadata>),
+    /// [List](https://infra.spec.whatwg.org/#list) (array)
     List(Vec<Metadata>),
+    /// [Boolean](https://infra.spec.whatwg.org/#boolean)
     Boolean(bool),
+    /// [Null](https://infra.spec.whatwg.org/#nulls)
     Null,
 }
 
+/// [DID Resolution Options](https://www.w3.org/TR/did-core/#did-resolution-options).
+///
+/// Formerly known as "DID resolution input metadata".
```

Comment:
should the tzkt option be mentioned here as an example of Additional options?

Reply:
It could be. It's specific to a particular DID method (did:tz / DIDTz). But this would be a place where it can be passed for that DID method/resolver.

Reply:
I hesitate to document that option here (and the other two did:tz options), as they are implementation/method-specific, and ssi is supposed to be de-coupled from particular DID methods. Proposing to add them to the DIDTz method/resolver implementation: https://github.com/spruceid/ssi/pull/357

For discovery of the options on the CLI, I think something else/more may be needed, e.g. updating the didkit CLI did:tz examples in https://github.com/spruceid/did-tezos/ README, or adding a more advanced CLI help system to introspect the available DID resolution options (and/or hard-coding some listing. cf. [method-pattern](https://github.com/spruceid/didkit/pull/217))